integrations/gitlab/gitlab_integration/gitlab_service.py

```diff
@@ -533,6 +550,84 @@ async def get_all_issues(self, group: Group) -> typing.AsyncIterator[List[Issue]
             issues: List[Issue] = typing.cast(List[Issue], issues_batch)
             yield issues
 
+    async def get_member_groups(
+        self, member: User
+    ) -> typing.AsyncIterator[List[Group]]:
+
+        semaphore = asyncio.Semaphore(MAX_CONCURRENT_TASKS)
+
+        async def check_group_membership(group: Group) -> Group | None:
+            "check if the user is a member of the group"
+            async with semaphore:
+                try:
+                    await AsyncFetcher.fetch_single(group.members.get, member.get_id())
+                    return group
+                except GitlabError as err:
+                    if err.response_code != 404:
+                        raise err
+                    return None
```

Comment:
this feels very insufficient, eventually for all the if I have 1 member, and 1000 groups, I'll have to query this api 1000 times to find out what groups he related to? isn't there any other way to get that data?